### ISSUE TITLE
Bogus error message on `List[str]`

### ISSUE THREAD

**Issue by rebcabin** [lcompilers/lpython#1944]
repro : https://github.com/rebcabin/lpython/tree/brian-lasr/lasr/lpython/Issue1944

The following works in ordinary CPython (after uncommenting the compatibility block at the top of the file):

```python
WHITESPACE   : List[str] = [" ", ",", NEWLINE, "\r", "\t"]
```

The error message from LPython does not lead me to a fix:

```bash
└─(15:12:05 on brian-lasr ✹ ✭)──> ~/CLionProjects/lpython/src/bin/lpython -I. lasr_lexer.py && ./lasr_lexer.out                                       2 ↵ ──(Sun,Jun18)─┘
semantic error: Variable 'str' not declared
   --> lasr_lexer.py:133:21
    |
133 | WHITESPACE   : List[str] = [" ", ",", NEWLINE, "\r", "\t"]
    |                     ^^^ 
```

**Comment by rebcabin** [comment]
There is a workaround: `list` instead of `List`. I incorporated the workaround and moved on to Issue #1945.

**Comment by Smit-create** [comment]
https://docs.python.org/3/library/typing.html#typing.List. `List` is a deprecated alias to `list`. The error message needs to be improved for sure.

**Comment by Thirumalai-Shaktivel** [comment]
I think the error message should suggest the user to use `list` instead of `List`.

**Comment by Smit-create** [comment]
Is this good to close this now?

**Comment by certik** [comment]
The error message is now:
```
semantic error: Unsupported type annotation: List
 --> a.py:1:16
  |
1 | WHITESPACE   : List[str] = [" ", ",", NEWLINE, "\r", "\t"]
  |                ^^^^^^^^^ 
```
Which is better, but not perfect. See #1993, after which it would say something like:
```
semantic error: The type `List` is undeclared
 --> a.py:1:16
  |
1 | WHITESPACE   : List[str] = [" ", ",", NEWLINE, "\r", "\t"]
  |                ^^^^^^^^^ 
```
Which I think is a good enough error message here. So closing this issue, the remaining issue to fix is #1993.
